x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -87,6 +46,12 @@ export interface ProviderLoginAttempt {
    */
   provider: Pick<AuthenticationProvider, 'name'> | Pick<AuthenticationProvider, 'type'>;
 
+  /**
+   * Optional URL to redirect user to after successful login. This URL is ignored if provider
+   * decides to redirect user to another URL after login.
+   */
+  redirectURL?: string;
```

Comment:
__note:__ currently `redirectURL` story is a bit confusing, we pass it to the Authenticator so that it knows where to finally redirect user to after Overwritten Session and Access Agreement UI and we also pass it to OIDC/SAML providers directly so that they can store it in the state before redirecting to IdP. In the future only Authenticator will need to deal with this in a generic fashion, but I don't know how to do that yet in a more or less elegant way :)